### DISCUSSION TITLE
Priority in the execution of Expressions

### DISCUSSION THREAD

**Discussion by jncengineering** [microsoft/RulesEngine#126] [Q&A]
Is it possible to include priorities in the execution of expressions?

**Comment by abbasc52** [comment]
HI @jncengineering can provide an example on priority of execution?

**Reply by inishantmishra** [reply]
@abbasc52 
Suppose we have 1 workflow with 4 rules and on some input, it matches 2 rules with isSuccess=true. Now I want to return 2nd rule because 2nd rule has the most priority then how can I choose 2nd one??
Currently, it gives us the first one from the list in the onSuccess event.

**Comment by bmasotta-lcg** [comment]
I found that rules are evaluated by name. Also all the rules are evaluated first and then all the actions are executed

The more natural way for doing that you want is to use nested rules with an "or" operator (not verified) and the NestedRuleExecutionMode = NestedRuleExecutionMode.Performance,
